### PR TITLE
Fixup all example CI tests and properly fail

### PR DESCRIPTION
**Pull request by muellerzr**
# Fix all example CI

## What does this add?

This pr makes a variety of fixes to the example tests to solve all failures and properly ensure that failures occur when they need to

## Why is it needed?

The entire list of issues I noticed are below:

- We cannot mock `CometMLTracker` to run offline tests with the tracking example script, so the CI now explicitly uninstalls them
- Adds a new `requires_tracking` decorator for the example tests to make sure at least one tracking API is installed and `comet_ml` is not installed
- Creates a new exception class and a new `run_command` function which will properly call a subprocess test function and check its outputs properly as well as fully fail if the call did not work.
- Fixes two bugs in the cross validation example in regards to noting the final prediction score.

### CODE REVIEW

File: examples/by_feature/cross_validation.py [hunk 1]
```diff
@@ -229,20 +229,14 @@ def training_function(config, args):
             with torch.no_grad():
                 outputs = model(**batch)
             predictions = outputs.logits
-            predictions, references = accelerator.gather((predictions, batch["labels"]))
```
Comment (sgugger) [inline examples/by_feature/cross_validation.py:232]: This will stop working on distributed setup. The predictions will need to be gathered (maybe not the labels).

**Review by sgugger — APPROVED** [review]
Thanks for all the fixes!

File: examples/by_feature/cross_validation.py (hunk 1 above)
Comment (muellerzr) [reply examples/by_feature/cross_validation.py:232]: Should be fixed now

**Comment by muellerzr** [comment]
All slow ci is now passing here: https://github.com/huggingface/accelerate/runs/7358153325

**Review by sgugger — APPROVED** [review]
